### PR TITLE
[WIP] Trieprefetch on stateprefetcher

### PR DESCRIPTION
**Pull request by qinglin89**
### Description

add a description of your changes here...

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: core/state/trie_prefetcher.go [hunk 1]
```diff
@@ -165,20 +168,61 @@ func (p *triePrefetcher) copy() *triePrefetcher {
 	}
 	return copy
 }
+func (p *triePrefetcher) cpy4Prefetcher() *triePrefetcher {
+	cpy := &triePrefetcher{
+		db:           p.db,
+		root:         p.root,
+		fetchers:     p.fetchers,
```
Comment (brilliant-lx) [inline core/state/trie_prefetcher.go:175]: the map will have concurrent access issue.
read by several `triePrefetcher` in different routine, and updated in prefetchLoop.

Comment (qinglin89) [reply core/state/trie_prefetcher.go:175]: won't have concurrency access issue, access would be managed by a shared channel with the same loop

File: core/state_prefetcher.go [hunk 2]
```diff
@@ -153,5 +153,10 @@ func precacheTransaction(msg types.Message, config *params.ChainConfig, gaspool
 	// Update the evm with the new transaction context.
 	evm.Reset(NewEVMTxContext(msg), statedb)
 	// Add addresses to access list if applicable
-	ApplyMessage(evm, msg, gaspool)
+	if _, err := ApplyMessage(evm, msg, gaspool); err != nil {
+		return
+	}
+	if config.IsByzantium(header.Number) {
+		statedb.Finalise4Prefetcher(true)
+	}
```
[the remark this answers is not in the document]
Comment (qinglin89) [reply core/state_prefetcher.go:161]: IsByzantium is used to be consistent with the system behavior
`Finalise4Prefetcher` will clear the journal.

File: core/state/statedb.go [hunk 3]
```diff
@@ -960,6 +973,27 @@ func (s *StateDB) Finalise(deleteEmptyObjects bool) {
 	// Invalidate journal because reverting across transactions is not allowed.
 	s.clearJournalAndRefund()
 }
+func (s *StateDB) Finalise4Prefetcher(deleteEmptyObjects bool) {
```
[the remark this answers is not in the document]
Comment (qinglin89) [reply core/state/statedb.go:976]: we just want to start the prefetcher, no need to deal with the pendingStorage.

File: core/state/trie_prefetcher.go (hunk 1 above)
Comment (brilliant-lx) [reply core/state/trie_prefetcher.go:175]: yes, you are right

**Pull request closed without merging**